docs/src/features.md

```diff
@@ -0,0 +1,136 @@
+# Miscellaneous Features
+
+In this tutorial we cover some of the features `QXTools` has to improve quantum circuit simulations and how to specify the type of output we would like our simulations to generate.
+
+## Hypergraphs
+
+One feature that `QXTools` uses by default is that of [hypergraphs](https://arxiv.org/abs/1710.05867). The hypergraph for a given tensor network is a graph, representing the network topology, whose nodes correspond to tensors in the network and edges correspond to indices in the network that connect different tensors. Note, two indices in the netwrok may be associated with the same edge in the hypergraph if they both connect to a diagonal tensor and are identical as a result. The tensor network topology can also be represented using an ordinary graph but it typically requires a greater number of edges to do so.
+
+This is useful because to find an efficient contraction plan for a given tensor network, `QXTools` first creates a line graph for the network topology and then runs the FlowCutter algorithm to find a tree decompostion for that line graph which can be converted into a contraction plan. Running FlowCutter on smaller graphs often results in more efficient contraction plans, with a smaller treewidth, in a shorter amount of time.
+
+`QXTools`, together with `QXTns`, can automatically detect the diagonal structure of gate tensors in a network and create the line graph of a network while taking the hypergraph structure of the network into account. We illustrate this below by creating the line graph for a network both with and without taking the hypergraph structure into account and show that using the hypergraph structure results in a smaller line graph.
+
+```
+# Create a random circuit.
+tnc = TensorNetworkCircuit(3)
+CX = [[1., 0., 0., 0.] [0., 1., 0., 0.] [0., 0., 0., 1.] [0., 0., 1., 0.]]
+
+for layer = 1:10
+    push!(tnc, [2, 1], CX)
+    push!(tnc, [2, 3], CX)
+    push!(tnc, [2], rand(2, 2))
+end
+
+# Create the line graph for the circuit without using the hypergraph.
+lg, symbol_map = convert_to_line_graph(tnc, use_hyperedges=false)
+@show lg
+
+# Create the line graph for the circuit using the hypergraph.
+lg, symbol_map = convert_to_line_graph(tnc, use_hyperedges=true)
+@show lg
```

Comment:
Maybe a sentence or two about the LabeledGraph data structure, i.e. that it is defined in QXGraphDecompositions and wraps a SimpleGraph (?) from LightGraphs.jl